docs/api-reference.rst

```diff
@@ -0,0 +1,246 @@
+        data collected is first passed through the :ref:`output processors
+        <loaders-processors>` to get the final value to assign to each
+        item field.
+
+    .. method:: nested_xpath(xpath)
+
+        Create a nested loader with an xpath selector.
+        The supplied selector is applied relative to selector associated
+        with this :class:`ItemLoader`. The nested loader shares the item
+        with the parent :class:`ItemLoader` so calls to :meth:`add_xpath`,
+        :meth:`add_value`, :meth:`replace_value`, etc. will behave as expected.
+
+    .. method:: nested_css(css)
+
+        Create a nested loader with a css selector.
+        The supplied selector is applied relative to selector associated
+        with this :class:`ItemLoader`. The nested loader shares the item
+        with the parent :class:`ItemLoader` so calls to :meth:`add_xpath`,
+        :meth:`add_value`, :meth:`replace_value`, etc. will behave as expected.
+
+    .. method:: get_collected_values(field_name)
+
+        Return the collected values for the given field.
+
+    .. method:: get_output_value(field_name)
+
+        Return the collected values parsed using the output processor, for the
+        given field. This method doesn't populate or modify the item at all.
+
+    .. method:: get_input_processor(field_name)
+
+        Return the input processor for the given field.
+
+    .. method:: get_output_processor(field_name)
+
+        Return the output processor for the given field.
+
+    :class:`ItemLoader` instances have the following attributes:
+
+    .. attribute:: item
+
+        The item object being parsed by this Item Loader.
+        This is mostly used as a property so when attempting to override this
+        value, you may want to check out :attr:`default_item_class` first.
+
+    .. attribute:: context
+
+        The currently active :ref:`Context <loaders-context>` of this
+        Item Loader.
+
+    .. attribute:: default_item_class
+
+        An Item class (or factory), used to instantiate items when not given in
+        the ``__init__`` method.
+
+    .. attribute:: default_input_processor
+
+        The default input processor to use for those fields which don't specify
+        one.
+
+    .. attribute:: default_output_processor
+
+        The default output processor to use for those fields which don't specify
+        one.
+
+    .. attribute:: default_selector_class
+
+        The class used to construct the :attr:`selector` of this
+        :class:`ItemLoader`, if only a response is given in the ``__init__`` method.
+        If a selector is given in the ``__init__`` method this attribute is ignored.
+        This attribute is sometimes overridden in subclasses.
+
+    .. attribute:: selector
+
+        The :class:`~parsel.Selector` object to extract data from.
+        It's either the selector given in the ``__init__`` method or one created from
+        the response given in the ``__init__`` method using the
```

Comment:
there is no response argument

Reply:
We would need to be careful with docstrings like that in Scrapy docs; in Scrapy docs it'd be nice to mention response in documentation for this atribute.